File: Qt_turtle_service/demo/ROS_demo/msg_demo/scripts/student_subs.py
#! /usr/bin/env python
# coding:utf-8

import rospy
from msg_demo.msg import Student


def subs_callback(msg):
    print ("姓名:{}, 年龄:{}, 身高:{}".format(msg.name, msg.age, msg.height))


if __name__ == "__main__":

    subscriber_node = 'person_sub'
    rospy.init_node(subscriber_node)

    topic_name = 'person_msgs'
    subs = rospy.Subscriber(topic_name, Student, subs_callback)

    rospy.spin()
    
    

File: ROS_demo/msg_demo/src/student_subs.cpp
// coding:utf-8

#include "ros/ros.h"
#include "msg_demo/Student.h"

void subs_callback(const msg_demo::Student::ConstPtr&people_p)
{
    ROS_INFO("订阅的人信息:%s, %d, %.2f", people_p->name.c_str(), people_p->age, people_p->height);
}

int main(int argc, char *argv[])
{
    /* code */
    setlocale(LC_ALL, "");

    ros::init(argc, argv, "people_subs");

    ros::NodeHandle nh;
    
    ros::Subscriber sub = nh.subscribe<msg_demo::Student>("people_msgs", 1000, subs_callback);

    ros::spin();


    return 0;
}
